### PR TITLE
Hyundai: Car Port for Azera 6th Generation

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with stock system: `87d7f06ade479c2e|2023-09-11--19-58-49`
- [x] route with openpilot: `87d7f06ade479c2e|2023-09-11--23-30-11`
- [x] harness Type: [Hyundai K](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)
- [x] architecture: CAN

**Link to all physical parts during installation**
  - [comma 3X](https://comma.ai/shop/comma-3x)
  - [car harness kit](https://comma.ai/shop/car-harness)
    - [Hyundai K](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)

Thanks to community 2022 Hyundai Azera owner al_lajlaaj_02124 (Discord).

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -1940,6 +1942,23 @@ def match_fw_to_car_fuzzy(live_fw_versions) -> Set[str]:
...
+    (Ecu.fwdCamera, 0x7c4, None): [
+      b'\xf1\x00IG  MFC  AT MES LHD 1.00 1.04 99211-G8100 200511',
+    ],
+    (Ecu.transmission, 0x7e1, None): [
+      b'\xf1\x00bcsh8p54  U912\x00\x00\x00\x00\x00\x00SIG0M35MH0\xa4 |.',
+    ],
+    (Ecu.engine, 0x7e0, None): [
+      b'\xf1\x81641KA051\x00\x00\x00\x00\x00\x00\x00\x00',
+    ],
+  },
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:1961]: No ABS?

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ class CAR:
   TUCSON_4TH_GEN = "HYUNDAI TUCSON 4TH GEN"
   TUCSON_HYBRID_4TH_GEN = "HYUNDAI TUCSON HYBRID 4TH GEN"
   SANTA_CRUZ_1ST_GEN = "HYUNDAI SANTA CRUZ 1ST GEN"
+  AZERA_2022 = "HYUNDAI AZERA 2022"
```
[the remark this answers is not in the document]
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py:102]: Looks like it's 7th Gen, will change to that. https://en.wikipedia.org/wiki/Hyundai_Grandeur#Seventh_generation_(GN7;_2022)

File: selfdrive/car/hyundai/values.py (hunk 1 above)
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py:1961]: Yep, no ABS FW.

Comment (sshane) [reply selfdrive/car/hyundai/values.py:1961]: Know why?

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py]: Not really sure, did a quick search and this car does have ABS (whew).

Comment (sshane) [reply selfdrive/car/hyundai/values.py]: It's definitely there, it just requires a different query `iso-tp query bad response: (2001, None) - 0x7f2231`.

**Comment by sshane** [comment]
All good to go?

**Comment by haram-KONA** [comment]
The vehicle called "Azera", called "Grandeur" in Korea, is Hyundai Motor Company's flagship vehicle and a representative vehicle in Korea. 
In the past, it was also popular in North America, but it doesn't seem to be used much in North America these days. 
If this vehicle is officially supported, a significant number of users in Korea will participate. 
It is Hyundai Motor Company's representative vehicle with a very large user base.
I always wondered why this vehicle wasn't officially supported yet.
This is a vehicle with a very large user base in Korea.
To date, approximately 3 million units have been sold since it was first released, 
and it is still the undisputed number one vehicle in the sedan segment in Korea.
In many ways, it seems like a really good thing to be providing official support for this vehicle.

Thanks You Sunny ~~~  and Comma.ai  ...

Only now is Korea's truly representative vehicle officially supported. !!!!

File: selfdrive/car/hyundai/hyundaican.py [hunk 3]
```diff
@@ -37,7 +37,8 @@ def create_lkas11(packer, frame, car_fingerprint, apply_steer, steer_req,
                          CAR.IONIQ_EV_2020, CAR.IONIQ_PHEV, CAR.KIA_SELTOS, CAR.ELANTRA_2021, CAR.GENESIS_G70_2020,
                          CAR.ELANTRA_HEV_2021, CAR.SONATA_HYBRID, CAR.KONA_EV, CAR.KONA_HEV, CAR.KONA_EV_2022,
                          CAR.SANTA_FE_2022, CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022,
-                         CAR.SANTA_FE_PHEV_2022, CAR.KIA_STINGER_2022, CAR.KIA_K5_HEV_2020, CAR.KIA_CEED):
+                         CAR.SANTA_FE_PHEV_2022, CAR.KIA_STINGER_2022, CAR.KIA_K5_HEV_2020, CAR.KIA_CEED,
+                         CAR.AZERA_7TH_GEN):
```
Comment (sshane) [inline selfdrive/car/hyundai/hyundaican.py:41]: It faults without this?

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/hyundaican.py:41]: Yes.

**Comment by sunnyhaibin** [comment]
> Is this only for UAE cars? Or is there a US version?

In Korea, the model is named Grandeur. Azera is used in other markets including UAE. The last model year that was sold in the US was 2017, but it seems like the 2022 model year is making a comeback per speculation.

File: selfdrive/car/hyundai/values.py [hunk 4]
```diff
@@ -209,6 +210,7 @@ def init_make(self, CP: car.CarParams):
   ],
   CAR.TUCSON_HYBRID_4TH_GEN: HyundaiCarInfo("Hyundai Tucson Hybrid 2022-23", "All", car_parts=CarParts.common([CarHarness.hyundai_n])),
   CAR.SANTA_CRUZ_1ST_GEN: HyundaiCarInfo("Hyundai Santa Cruz 2022-23", car_parts=CarParts.common([CarHarness.hyundai_n])),
+  CAR.AZERA_7TH_GEN: HyundaiCarInfo("Hyundai Azera (Middle East only) 2022", "All", car_parts=CarParts.common([CarHarness.hyundai_k])),
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/values.py:213]: no need to say "middle east only" if it's only sold in the middle east

File: selfdrive/car/hyundai/values.py [hunk 5]
```diff
@@ -1940,6 +1942,23 @@ def match_fw_to_car_fuzzy(live_fw_versions) -> Set[str]:
       b'\xf1\x00MQhe SCC FHCUP      1.00 1.07 99110-P4000         ',
     ],
   },
+  CAR.AZERA_7TH_GEN: {
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:1945]: This is actually the 6th gen, you can tell by the platform code.

image

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py]: This is very odd.

Comment (sshane) [reply selfdrive/car/hyundai/values.py]: What is odd?

**Pull request merged**